testing/test_assertion.py

```diff
@@ -546,6 +547,146 @@ def test_mojibake(self):
         assert msg
 
 
+class TestAssert_reprcompare_dataclass(object):
+    @pytest.mark.skipif(sys.version_info < (3, 7), reason="Dataclasses in Python3.7+")
+    def test_dataclasses(self):
+        from dataclasses import dataclass, field
+
+        @dataclass
+        class SimpleDataObject(object):
+            field_a = field()
```

Comment:
I initially used type annotations here, but tests failed for 2.7 so decided to use `field` as we only care about output, not the type.

Reply:
Looks like 3.7 requires type annotation. Thoughts on a good way of getting both in there?

I can edit the class dict's `__annotations__` item